pkg/apis/servicecatalog/types.go

```diff
@@ -294,23 +336,26 @@ type CommonServiceClassSpec struct {
 	Requires []string
 }
 
-// ClusterServiceClassSpec represents the details about a ClusterServiceClass
+// ClusterServiceClassSpec represents the details about a ClusterServiceClass.
 type ClusterServiceClassSpec struct {
 	CommonServiceClassSpec
 
-	// ClusterServiceBrokerName is the reference to the Broker that provides this
-	// ClusterServiceClass.
+	// ClusterServiceBrokerName is the reference to the ClusterServiceBroker that
+	// provides this ClusterServiceClass.
 	//
 	// Immutable.
 	ClusterServiceBrokerName string
 }
 
-// ClusterServiceClassStatus represents status information about a
-// ClusterServiceClass.
-type ClusterServiceClassStatus struct {
-	// RemovedFromBrokerCatalog indicates that the broker removed the service from its
-	// catalog.
-	RemovedFromBrokerCatalog bool
+// ServiceClassSpec represents the details about a ServiceClass.
+type ServiceClassSpec struct {
+	CommonServiceClassSpec
+
+	// ServiceBrokerName is the reference to the ServiceBroker that provides this
+	// ServiceClass.
+	//
+	// Immutable.
+	ServiceBrokerName string
```

Comment:
Seeing this makes me wonder if we should make a convention to share this property. like make it an object:

```
type ServiceClassSpec struct {
  ...
  ServiceBroker ServiceBrokerRef
  ...
}

type ServiceBrokerRef struct {
  Type Type
  Name string
}
```

And then it could look like:

`clusterServiceClass.Spec.ServiceBroker.Name -> MyBroker`
`clusterServiceClass.Spec.ServiceBroker.Type -> ClusterServiceBroker`

I am not requesting changes, just thinking. Maybe others will say something. It could always be something we refactor towards.

Reply:
I am trying to think of a way so that the name is always a consistent path in the object. It could make writing the way the reconciliation loop code becomes common/generic for cluster/ns versions of classes more similar.

Reply:
We can't make a backward incompatible change to the existing API.